src/lib.rs

```diff
@@ -0,0 +1,92 @@
+mod cli;
+pub mod config;
+mod types;
+
+use crate::config::Config;
+use actix_web::{
+    error,
+    web::{self, Json},
+    App, Error, HttpServer,
+};
+use std::{io::Write, path::PathBuf, process::Command};
+use types::{SolToUmlRequest, SolToUmlResponse};
+use uuid::Uuid;
+
+async fn sol_to_uml(
+    tmp_dir: web::Data<PathBuf>,
+    data: Json<SolToUmlRequest>,
+) -> Result<Json<SolToUmlResponse>, Error> {
+    let data = data.into_inner();
+
+    let unique_name = Uuid::new_v4().to_string();
+    let contract_dir = tmp_dir.get_ref().join(&unique_name);
+    let contract_dir =
+        web::block(move || std::fs::create_dir(contract_dir.clone()).map(|_| contract_dir))
+            .await
+            .map_err(error::ErrorInternalServerError)??;
+
+    for (name, content) in data.sources.iter() {
+        let file_path = contract_dir.join(name.as_path());
+        let prefix = file_path.parent();
+        if let Some(prefix) = prefix {
+            let prefix = prefix.to_path_buf();
+            web::block(move || std::fs::create_dir_all(prefix))
+                .await
+                .map_err(error::ErrorBadRequest)??;
+        }
+
+        let mut f = web::block(move || std::fs::File::create(file_path))
+            .await
+            .map_err(error::ErrorBadRequest)??;
+        let content = (*content).clone();
+        web::block(move || f.write_all(content.as_bytes()))
+            .await
+            .map_err(error::ErrorBadRequest)??;
+    }
+
+    let uml_path = contract_dir.join(format!("{unique_name}.svg"));
+    let status = Command::new("sol2uml")
+        .arg(&contract_dir)
+        .arg("-o")
+        .arg(&uml_path)
+        .status()
+        .expect("sol2uml command failed to start");
```

Comment:
Here it make sense to use `web::block`, as that may require some time to complete and may block the whole thread.

Also you may leave current variables into the scope via:
```
let status = {
    let contract_dir = contract_dir.clone();
    let uml_path = uml_path.clone();
    ...
}
```

Reply:
Yes, fixed it